test/e2eprovider/server/server_test.go

```diff
@@ -0,0 +1,141 @@
+// +build e2e
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
+)
+
+var (
+	testMockServer *Server
+	tempDir        = os.TempDir()
+)
+
+func TestMain(m *testing.M) {
+	setup()
+	exitCode := m.Run()
+	teardown()
+	os.Exit(exitCode)
+}
+
+func setup() {
+	var err error
+
+	testMockServer, err = NewE2EProviderServer(fmt.Sprintf("unix://%s/%s", tempDir, "e2e-provider.sock"))
+	if err != nil {
+		panic(err)
+	}
+}
+
+func teardown() {
+	testMockServer.Stop()
+	os.RemoveAll(tempDir)
+}
+
+func TestMockServer(t *testing.T) {
+	cases := []struct {
+		name string
+	}{
+		{
+			name: "start",
```

Comment:
IMO we don't need table tests here as it's just checking the server start.

Reply:
Yes.. But I thought to be consistent with other tests. Also easier to read test result after running test on local or in pipeline. WDYT?